Classes/FrontendEnvironment/Tsfe.php

```diff
@@ -168,10 +168,10 @@ protected function initializeTsfe(int $pageId, int $language = 0, ?int $rootPage
             $backedUpBackendUser = $GLOBALS['BE_USER'] ?? null;
             try {
                 $tsfe->determineId($serverRequest);
-                $serverRequest->withAttribute('frontend.controller', $tsfe);
+                $serverRequest = $serverRequest->withAttribute('frontend.controller', $tsfe);
```

Comment:
Maybe we should switch the lines, to provide the same instance of `$serverRequest`?

Reply:
Not really sure if that's needed. In the [`TypoScriptFrontendInitialization`](https://github.com/TYPO3/typo3/blob/12.4/typo3/sysext/frontend/Classes/Middleware/TypoScriptFrontendInitialization.php#L92) Middleware `determineId()` is also called before adding the `frontend.controller` attribute to the request.